app/client/components/addLecture/LectureDurationSelect.tsx

```diff
@@ -0,0 +1,67 @@
+import { zosiaApi, zosiaApiRoutes } from "@client/utils/zosiaApi";
+import { Select } from "@headlessui/react";
+import { useQuery } from "@tanstack/react-query";
+import React from "react";
+import { WidgetHandler } from "reactivated/dist/forms";
+import { DjangoFormsWidgetsSelect } from "reactivated/dist/generated";
+
+interface LectureDurationSelectProps {
+  field: WidgetHandler<DjangoFormsWidgetsSelect>;
+  lectureType: string;
+}
+
+interface LectureDurations {
+  durations: string[];
+}
+
+export const LectureDurationSelect = ({
+  field,
+  lectureType,
+}: LectureDurationSelectProps) => {
+  const { isPending, isError, data, error } = useQuery({
+    queryKey: [zosiaApiRoutes.addLectureDurations, lectureType],
+    queryFn: async ({ queryKey }) => {
+      const lecture = queryKey[1];
+      if (lecture === "") {
```

Comment:
We should validate outside of the query function; the query function is only for calling an API.

Reply:
In this case I see it more like this:

If the user didn't choose a lecture type, then I know what the API should return (that is - nothing), so instead of calling it, I will simply return some placeholder value immediately.